src/RequestHandlerSwooleRunner.php

```diff
@@ -356,14 +382,58 @@ private function getStaticResource(
         SwooleHttpRequest $request,
         SwooleHttpResponse $response
     ) : bool {
-        $staticFile = $this->docRoot . $request->server['request_uri'];
+        $server = $request->server;
+        if ($server['request_method'] !== 'GET' && $server['request_method'] !== 'HEAD') {
+            return false;
+        }
+        $staticFile = $this->docRoot . $server['request_uri'];
         if (! isset($this->cacheTypeFile[$staticFile])
             && ! $this->cacheFile($staticFile)
         ) {
             return false;
         }
 
-        $response->header('Content-Type', $this->cacheTypeFile[$staticFile]);
+        $response->header('Content-Type', $this->cacheTypeFile[$staticFile], true);
+
+        // Handle ETag and Last-Modified header
+        $lastModifiedTime = filemtime($staticFile) ?? 0;
+        $fileSize = filesize($staticFile) ?? 0;
+
+        // Set Last-Modified header
+        $lastModifiedTimeDateFormatted = trim(gmstrftime('%A %d-%b-%y %T %Z', $lastModifiedTime));
+        $response->header('Last-Modified', $lastModifiedTimeDateFormatted, true);
+
+        // Set ETag header
+        $etag = '';
+        switch ($this->etagValidationType) {
+            case self::WEAK_VALIDATION:
+                if ($lastModifiedTime && $fileSize) {
+                    $etag = 'W/"' . dechex($lastModifiedTime) . '-' . dechex($fileSize) . '"';
```

Comment:
You asked about algorithms... what you have here is almost identical to what nginx generates: https://serverfault.com/a/690374

:+1: